core/app/[locale]/(default)/account/(tabs)/addresses/page.tsx

```diff
@@ -43,7 +49,13 @@ export default async function Addresses({ searchParams }: Props) {
   return (
     <>
       <TabHeading heading="addresses" />
-      <AddressBook addressesCount={addressesCount} customerAddresses={addresses} key={endCursor}>
+      <AddressBook 
+        addressesCount={addressesCount}
+        customerAddresses={addresses}
+        key={endCursor}
+        pageInfo={pageInfo}
```

Comment:
Small suggestion here - we use key for [resetting](https://react.dev/reference/react/useState#resetting-state-with-a-key) state. So if we do component's refactoring then we can go further and optimize its props in the next way.

on deleting all addresses on a Page we check if there a Previous page and if so we move there:
```
if (addressBook.length === 0 && hasPreviousPage) {
      router.push(`/account/addresses/?before=${startCursor}`);
    }
```
So basically we just need `hasPreviousPage` & `startCursor` from `PageInfo` data. Based on this we can add them to `AddressBook` Props directly and remove `key` and `PageInfo` instead. Maybe I missed smth, wdyt?

Reply:
I optimized the props passing, but removing the `key` breaks the pagination, so I left it to reset the component state when switching between pages. Thanks for the suggestion!